### PR TITLE
NOISSUE: Add provision service

### PR DESCRIPTION
**Pull request by mteodor**
Pull request title should be `MF-XXX - description` or `NOISSUE - description` where XXX is ID of issue that this PR relate to.
Please review the [CONTRIBUTING.md](./CONTRIBUTING.md) file for detailed contributing guidelines.

### Which issue(s) does this PR fix/relate to?
Put here `Resolves #XXX` to auto-close the issue that your PR fixes (if such)

### CODE REVIEW

File: provision/LICENSE [hunk 1]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (drasko) [inline provision/LICENSE]: Should be removed.

**Review by drasko — CHANGES_REQUESTED** [review]

File: cmd/provision/main.go [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+		},
+		Bootstrap: provision.Bootstrap{
+			X509Provision: provisionX509,
+			Provision:     provisionBS,
+			AutoWhiteList: autoWhiteList,
+			Content:       fmt.Sprintf(mainflux.Env(envBSContent, defBSContent), mainflux.Env(envMQTTURL, defMQTTURL)),
+		},
+
+		// This is default conf for provision if there is no
+		// config file
```
Comment (manuio) [inline cmd/provision/main.go]: IMO No need to split this comment

File: provision/README.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+PROVISION service provides an HTTP API to interact with Mainflux.
+
+## Configuration
+
+The service is configured using the environment variables presented in the
+following table. Note that any unset variables will be replaced with their
+default values.
+
+| Variable                            | Description                                       | Default                          |
+| ----------------------------------- | ------------------------------------------------- | -------------------------------- |
```
Comment (manuio) [inline provision/README.md:12]: Remove `""` from default values

**Review by manuio — CHANGES_REQUESTED** [review]

File: .env [hunk 4]
```diff
@@ -103,6 +103,28 @@ MF_BOOTSTRAP_DB_PASS=mainflux
 MF_BOOTSTRAP_DB=bootstrap
 MF_BOOTSTRAP_DB_SSL_MODE=disable
 
+### Provision
+MF_PROVISION_CONFIG_FILE=/configs/config.toml
+MF_PROVISION_LOG_LEVEL=info
+MF_PROVISION_HTTP_PORT=8190
+MF_PROVISION_ENV_CLIENTS_TLS=false
+MF_PROVISION_CA_CERTS=
```
Comment (drasko) [inline .env:107]: Do we leave it like this on other services or we put explicitly empty string, i.e. `""`?

File: provision/sdk/sdk.go [hunk 5]
```diff
@@ -0,0 +1,318 @@
+package sdk
```
Comment (drasko) [inline provision/sdk/sdk.go:1]: This should be added to Mainflux SDK, right? I do not see the point to have a separate SDK per service, we Mainflux SDK should be used for the whole system. We recently added Boostrap commands to it, this also should be transferred over there.

**Review by drasko — CHANGES_REQUESTED** [review]

File: .env (hunk 4 above)
Comment (mteodor) [reply .env:107]: yes

File: provision/sdk/sdk.go (hunk 5 above)
Comment (mteodor) [reply provision/sdk/sdk.go:1]: this package is provision specific wrapper around mainflux sdk

Comment (drasko) [reply provision/sdk/sdk.go:1]: Can it be useful for other applications, and if yes why not putting it available to public though one unified SDK library?

File: .env (hunk 4 above)
Comment (drasko) [reply .env:107]: Yes what?

**Comment by drasko** [comment]
Closing this one, please commit your changes to this branch: https://github.com/mainflux/mainflux/tree/provision and hen open new PR starting from this branch.

**Pull request closed without merging**